### PR TITLE
chore: remove openssl from process.versions

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Previously, we were leaving the OpenSSL version in Node.js in our `process.versions.openssl` since we set `node_use_openssl = true` in our `BUILD.gn` file for parity with Node.js, but this is confusing to users since that does not exist in Electron. This PR checks for `process.versions.openssl` being set, and then if it is, removes it. In the future, i'd like to actually show the BoringSSL version there instead, but I feel that can be done in a separate PR.

This also has the handy side effect of letting us remove a bunch of disabled Node.js crypto tests.

cc @MarshallOfSound @nornagon 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: shell/app/node_main.cc [hunk 1]
```diff
@@ -113,6 +113,10 @@ int NodeMain(int argc, char* argv[]) {
     gin_helper::Dictionary versions;
     if (process.Get("versions", &versions)) {
       versions.SetReadOnly(ELECTRON_PROJECT_NAME, ELECTRON_VERSION_STRING);
+
+      // We don't build with OpenSSL
+      if (versions.Has("openssl"))
+        versions.Delete("openssl");
```
Comment (poiru) [inline shell/app/node_main.cc]: Do we need `Has()`? I think `Delete()` should just no-ops if the key doesn't exist.

Comment (codebytere) [reply shell/app/node_main.cc]: oh i think you're right - will update.

**Comment by codebytere** [comment]
Do you see any possible path forward here? It would be nice to indicate to users that they are not actually getting OpenSSL in a more programmatic way, which is what this intended to accomplish.

**Comment by nornagon** [comment]
You can easily determine that you're not running openssl by checking `process.versions.electron`. I'm not aware of any other major Node-like environments that ship non-OpenSSL, so that seems like a reasonable way to check to me. Even better would be to check for the specific features that you need, since BoringSSL is _mostly_ compatible with OpenSSL.

**Pull request closed without merging**